### DISCUSSION TITLE
Consider replacing Guava Lists.newArrayList (and similar) usage?

### DISCUSSION THREAD

**Discussion by sleberknight** [kiwiproject/retrying-again#39] [General]
For _unmodifiable_ lists, sets, and maps, should we use Java's factory methods that were added in JDK 9 (collection factory methods)?

This would include replacing `Lists.newArrayList()` with `List.of()` and `Sets.newHashSet()` with `Set.of()`.

Obviously, if the list or set isn't _unmodifiable_ (i.e. elements are added after construction) then we can't make this change (without other design changes). In those cases, I'd recommend we stick with the Guava factory methods since you can "one-liner" create them with elements.

**Comment by chrisrohr** [comment]
I'm on board with this change. We should use native APIs as much as possible.

**Reply by sleberknight** [reply]
Added #42 (the answer) which makes this discussion closed.
